src/lib/images/proxy.mts

```diff
@@ -98,24 +123,39 @@ export const transformImageParams = function (query) {
     .join(',')
 }
 
-// @ts-expect-error TS(7031) FIXME: Binding element 'config' implicitly has an 'any' t... Remove this comment to see the full error message
-export const initializeProxy = async function ({ config }) {
+export const initializeProxy = async function ({
+  config,
+  settings,
+}: {
+  config: NetlifyConfig
+  settings: ServerSettings
+}) {
   const remoteDomains = await parseRemoteImageDomains({ config })
+  const devServerUrl = getProxyUrl(settings)
 
   const ipx = createIPX({
     storage: ipxFSStorage({ dir: config?.build?.publish ?? './public' }),
-    httpStorage: ipxHttpStorage({ domains: remoteDomains }),
+    httpStorage: ipxHttpStorage({ domains: [...remoteDomains, devServerUrl] }),
   })
 
   const handler = createIPXNodeServer(ipx)
   const app = express()
 
   app.use(IMAGE_URL_PATTERN, async (req, res) => {
     const { url, ...query } = req.query
-    const modifiers = await transformImageParams(query)
-    // @ts-expect-error TS(2345) FIXME: Argument of type 'string | string[] | ParsedQs | P... Remove this comment to see the full error message
-    const path = `/${modifiers}/${encodeURIComponent(url)}`
-    req.url = path
+    const sourceImagePath = url as string
+    const modifiers = (await transformImageParams(query)) || `_`
+    if (!sourceImagePath.startsWith('http://') && !sourceImagePath.startsWith('https://')) {
+      // Construct the full URL for relative paths to request from development server
+      const sourceImagePathWithLeadingSlash = sourceImagePath.startsWith('/') ? sourceImagePath : `/${sourceImagePath}`
+      const fullImageUrl = `${devServerUrl}/${encodeURIComponent(sourceImagePathWithLeadingSlash)}`
+      console.log(`fullImageUrl: ${fullImageUrl}`)
+      req.url = `/${modifiers}/${fullImageUrl}`
+    } else {
+      // If the image is remote, we can just pass the URL as is
+      req.url = `/${modifiers}/${encodeURIComponent(sourceImagePath)}`
+    }
+
```

Comment:
This the only other functional change, uses the dev server to get the image instead of relying on the local `dist` directory.